deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,244 @@
+            # Evaluate
+            curr_results_aug = evaluate(copy_data_loader)
+            results_aug_df = pd.DataFrame(curr_results_aug, columns=['Class', 'Metric', 'Value', 'N']
+                                          ).sort_values(by=['Class'])
+            # This modifies the augmentation name in-place
+            results_base_df["Augmentation"] = aug_name
+            # This adds the augmentation
+            results_aug_df["Augmentation"] = aug_name
+            results.append(pd.concat([results_base_df, results_aug_df], keys=["Baseline", "Augmented"]))
+
+            examples.append(self._get_random_image_pairs_from_dataloder(dataset.get_data_loader(), copy_data_loader))
+
+        # Create grand DataFrame from dictionary of augmentations
+        # This has flat hierarchy for augmentations and baseline-vs-augmentations
+        results_df = pd.concat(results, axis=0). \
+            reset_index(). \
+            rename({"level_0": "Status"}, axis=1). \
+            drop(labels="level_1", axis=1)
+
+        def _is_robust_to_augmentation(p, epsilon=10 ** -2):
+            grouped_results = p[["Status", "Metric", "Value"]].groupby(["Status", "Metric"]).median().reset_index()
+            differences = []
+            metric_statuses = []
+            for metric in grouped_results["Metric"].unique():
+                diff = grouped_results[grouped_results["Metric"] == metric]["Value"].diff().abs().iloc[-1]
+                differences.append(diff > epsilon)
+                metric_statuses.append(diff)
+            difference = any(differences)
+            return pd.Series([difference] + metric_statuses, index=["Affected"] +
+                                                                   [f"{m}_delta" for m in
+                                                                    grouped_results["Metric"].unique().tolist()])
+
+        # Iterating this dataframe per [Status, Augmentation] will give us easy comparisons
+        metric_results = results_df.groupby(["Augmentation"]).apply(_is_robust_to_augmentation,
+                                                                    self._epsilon).reset_index()
+
+        return metric_results, examples
+
+    def _get_random_image_pairs_from_dataloder(self, baseline_data_loader, augmented_data_loader, n_samples=10):
+        """
+        We iterate the internal dataset object directly to avoid randomness
+        Dataset returns data points as processed images, making this currently not really usable
+        To avoid making more assumptions this currently stays as-is
+        Note that images return in RGB format, ond to visualize them using OpenCV the final dimesion should be
+        transposed;
+        can be done via image = image[:, :, ::-1] or cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+        """
+        # We definitely make the assumption that the underlying structure is torch.utils.Dataset
+        baseline_sampler = iter(baseline_data_loader.dataset)
```

Comment:
in most datasets (or at least datasets which have `__getitem__`, and I suspet that even for Iterables) this would most certainly lead to sampling along the image order, which means it's likely to get images of the same class. We must have a method to select here, whether it is a way of sampling randomly, selecting samples of different classes or selecting the images for which the "loss" is greatest (from different classes).

Reply:
It depends if the data is shuffled or not. Anyway I'm not sure how to implement it without going over the data a lot of times